### ISSUE TITLE
No compiler output shown; VSCode extension crashing

### ISSUE THREAD

**Issue by joshhansen** [rust-analyzer/rust-analyzer#4086]
I believe this arose since the most recent update. I have numerous compiler warnings and errors as well as clippy lints in my code, but none appear in VSCode using the rust-analyzer extension. Further, the "Rust Analyzer Language Server" output shows apparent repeated crashes:

```
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'failed to parse path ModPath { kind: Crate, segments: [] }, `crate`', crates/ra_assists/src/ast_transform.rs:185:28
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: <unknown>
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <unknown>
  51: <unknown>
  52: <unknown>
  53: <unknown>
  54: <unknown>
  55: <unknown>
...
```

Happy to provide further detail.

**Comment by lnicola** [comment]
Can you please try the nightly version? It might have better stack traces.

**Comment by joshhansen** [comment]
I did `rustup install nightly; rustup default nightly" and relaunched, hopefully that is sufficient.
I now see this output:
[ERROR rust_analyzer::main_loop] overly long loop turn: 109.660652ms
I do see some warnings for one project in my workspace. But for the one with tons of errors, I see nothing.

**Comment by Veetaha** [comment]
What lnicola meant is nightly rust-analyzer extension. Try adding this to `settings.json`:
```json
{
    "rust-analyzer.updates.channel": "nightly"
}
```

**Comment by joshhansen** [comment]
Got it. Now using 0.3.145-nightly. The prior reported outputs are no longer visible, but the extension host has this:

```
[2020-04-22 15:00:44.456] [exthost] [info] extension host started
[2020-04-22 15:00:44.530] [exthost] [info] ExtensionService#_doActivateExtension vadimcn.vscode-lldb {"startup":false,"extensionId":{"value":"hdevalke.rust-test-lens","_lower":"hdevalke.rust-test-lens"},"activationEvent":"onLanguage:rust"}
[2020-04-22 15:00:44.531] [exthost] [info] ExtensionService#loadCommonJSModule file:///home/josh/.vscode/extensions/vadimcn.vscode-lldb-1.5.0/extension.js
[2020-04-22 15:00:44.537] [exthost] [info] ExtensionService#_doActivateExtension matklad.rust-analyzer {"startup":false,"extensionId":{"value":"matklad.rust-analyzer","_lower":"matklad.rust-analyzer"},"activationEvent":"onLanguage:rust"}
[2020-04-22 15:00:44.537] [exthost] [info] ExtensionService#loadCommonJSModule file:///home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main
[2020-04-22 15:00:44.548] [exthost] [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch {"startup":true,"extensionId":{"value":"vscode.debug-auto-launch","_lower":"vscode.debug-auto-launch"},"activationEvent":"*"}
[2020-04-22 15:00:44.548] [exthost] [info] ExtensionService#loadCommonJSModule file:///usr/share/code/resources/app/extensions/debug-auto-launch/dist/extension
[2020-04-22 15:00:44.549] [exthost] [info] ExtensionService#_doActivateExtension vscode.emmet {"startup":true,"extensionId":{"value":"vscode.emmet","_lower":"vscode.emmet"},"activationEvent":"*"}
[2020-04-22 15:00:44.550] [exthost] [info] ExtensionService#loadCommonJSModule file:///usr/share/code/resources/app/extensions/emmet/dist/extension
[2020-04-22 15:00:44.558] [exthost] [info] ExtensionService#_doActivateExtension vscode.git {"startup":true,"extensionId":{"value":"vscode.git","_lower":"vscode.git"},"activationEvent":"*"}
[2020-04-22 15:00:44.559] [exthost] [info] ExtensionService#loadCommonJSModule file:///usr/share/code/resources/app/extensions/git/dist/main
[2020-04-22 15:00:44.571] [exthost] [info] ExtensionService#_doActivateExtension vscode.github-authentication {"startup":true,"extensionId":{"value":"vscode.github-authentication","_lower":"vscode.github-authentication"},"activationEvent":"*"}
[2020-04-22 15:00:44.571] [exthost] [info] ExtensionService#loadCommonJSModule file:///usr/share/code/resources/app/extensions/github-authentication/dist/extension.js
[2020-04-22 15:00:44.572] [exthost] [info] ExtensionService#_doActivateExtension vscode.merge-conflict {"startup":true,"extensionId":{"value":"vscode.merge-conflict","_lower":"vscode.merge-conflict"},"activationEvent":"*"}
[2020-04-22 15:00:44.572] [exthost] [info] ExtensionService#loadCommonJSModule file:///usr/share/code/resources/app/extensions/merge-conflict/dist/extension
[2020-04-22 15:00:44.574] [exthost] [info] ExtensionService#_doActivateExtension vscode.search-result {"startup":true,"extensionId":{"value":"vscode.search-result","_lower":"vscode.search-result"},"activationEvent":"*"}
[2020-04-22 15:00:44.574] [exthost] [info] ExtensionService#loadCommonJSModule file:///usr/share/code/resources/app/extensions/search-result/dist/extension.js
[2020-04-22 15:00:44.575] [exthost] [info] ExtensionService#_doActivateExtension vscode.vscode-account {"startup":true,"extensionId":{"value":"vscode.vscode-account","_lower":"vscode.vscode-account"},"activationEvent":"*"}
[2020-04-22 15:00:44.575] [exthost] [info] ExtensionService#loadCommonJSModule file:///usr/share/code/resources/app/extensions/vscode-account/dist/extension.js
[2020-04-22 15:00:44.729] [exthost] [info] ExtensionService#_doActivateExtension hdevalke.rust-test-lens {"startup":false,"extensionId":{"value":"hdevalke.rust-test-lens","_lower":"hdevalke.rust-test-lens"},"activationEvent":"onLanguage:rust"}
[2020-04-22 15:00:44.729] [exthost] [info] ExtensionService#loadCommonJSModule file:///home/josh/.vscode/extensions/hdevalke.rust-test-lens-1.0.0/out/extension
[2020-04-22 15:00:44.953] [exthost] [info] eager extensions activated
[2020-04-22 15:00:45.825] [exthost] [error] [matklad.rust-analyzer] provider FAILED
[2020-04-22 15:00:45.827] [exthost] [error] Error: content modified
	at handleResponse (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3520:48)
	at processMessageQueue (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3345:17)
	at Immediate.<anonymous> (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3329:13)
	at processImmediate (internal/timers.js:439:21)
[2020-04-22 15:00:45.828] [exthost] [error] [matklad.rust-analyzer] provider FAILED
[2020-04-22 15:00:45.828] [exthost] [error] Error: content modified
	at handleResponse (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3520:48)
	at processMessageQueue (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3345:17)
	at Immediate.<anonymous> (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3329:13)
	at processImmediate (internal/timers.js:439:21)
[2020-04-22 15:00:45.828] [exthost] [error] [matklad.rust-analyzer] provider FAILED
[2020-04-22 15:00:45.828] [exthost] [error] Error: content modified
	at handleResponse (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3520:48)
	at processMessageQueue (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3345:17)
	at Immediate.<anonymous> (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3329:13)
	at processImmediate (internal/timers.js:439:21)
[2020-04-22 15:00:45.835] [exthost] [error] [matklad.rust-analyzer] provider FAILED
[2020-04-22 15:00:45.835] [exthost] [error] Error: content modified
	at handleResponse (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3520:48)
	at processMessageQueue (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3345:17)
	at Immediate.<anonymous> (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3329:13)
	at processImmediate (internal/timers.js:439:21)
[2020-04-22 15:00:45.971] [exthost] [error] [matklad.rust-analyzer] provider FAILED
[2020-04-22 15:00:45.971] [exthost] [error] Error: content modified
	at handleResponse (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3520:48)
	at processMessageQueue (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3345:17)
	at Immediate.<anonymous> (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3329:13)
	at processImmediate (internal/timers.js:439:21)
[2020-04-22 15:00:49.647] [exthost] [error] [matklad.rust-analyzer] provider FAILED
[2020-04-22 15:00:49.647] [exthost] [error] Error: content modified
	at handleResponse (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3520:48)
	at processMessageQueue (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3345:17)
	at Immediate.<anonymous> (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3329:13)
	at processImmediate (internal/timers.js:439:21)
[2020-04-22 15:00:49.647] [exthost] [error] [matklad.rust-analyzer] provider FAILED
[2020-04-22 15:00:49.647] [exthost] [error] Error: content modified
	at handleResponse (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3520:48)
	at processMessageQueue (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3345:17)
	at Immediate.<anonymous> (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3329:13)
	at processImmediate (internal/timers.js:439:21)
[2020-04-22 15:01:01.760] [exthost] [error] [matklad.rust-analyzer] provider FAILED
[2020-04-22 15:01:01.760] [exthost] [error] Error: content modified
	at handleResponse (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3520:48)
	at processMessageQueue (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3345:17)
	at Immediate.<anonymous> (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3329:13)
	at processImmediate (internal/timers.js:439:21)
[2020-04-22 15:01:01.760] [exthost] [error] [matklad.rust-analyzer] provider FAILED
[2020-04-22 15:01:01.760] [exthost] [error] Error: content modified
	at handleResponse (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3520:48)
	at processMessageQueue (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3345:17)
	at Immediate.<anonymous> (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3329:13)
	at processImmediate (internal/timers.js:439:21)
[2020-04-22 15:01:02.570] [exthost] [error] [matklad.rust-analyzer] provider FAILED
[2020-04-22 15:01:02.570] [exthost] [error] Error: content modified
	at handleResponse (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3520:48)
	at processMessageQueue (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3345:17)
	at Immediate.<anonymous> (/home/josh/.vscode/extensions/matklad.rust-analyzer-0.3.145-nightly/out/main.js:3329:13)
	at processImmediate (internal/timers.js:439:21)
```

I continue to see warnings from my project that compiles, but nothing from the one with errors and warnings.

**Comment by Veetaha** [comment]
@joshhansen
I'd recommend using a spoiler for a long wall of logs, just for the convenience of readers:
````
<details>
<summary>Spoiler title</summary>

```
Log goes here, also the new line after </summary> is important!
```
<details>
````

**Comment by lnicola** [comment]
I think "content modified" errors indicate cancellation of a request, which is normal during editing. Perhaps we shouldn't display them.

**Comment by Veetaha** [comment]
I can't deduce that this does come from our extension code by the given stack trace, this might rightfully be `vscode-languageclient`'s fault... If only I could reproduce...

**Comment by lnicola** [comment]
https://github.com/rust-analyzer/rust-analyzer/blob/3f1f3a835a07983344f6f3becf7a0c32b96d68bc/crates/rust-analyzer/src/main_loop.rs#L881

**Comment by lnicola** [comment]
Probably fixed by #6150.
